### PR TITLE
fix(Select): Fixed dot next to the spinner in grouped view more variant

### LINKED ISSUE #6383: bug(Select): Grouped options typeahead with loadingVariant: spinner shows some dot next to the spinner
**Describe the issue. What is the expected and unexpected behavior?**

See the dot next to the spinner.
Screen Shot 2021-09-30 at 18 34 22

**Please provide the steps to reproduce. Feel free to link CodeSandbox or another tool.**
https://codesandbox.io/s/hidden-meadow-kkj0q?file=/index.js

**Is this a bug or enhancement? If this issue is a bug, is this issue blocking you or is there a work-around?**
bug

### PR DESCRIPTION
**Pull request by tlabaj**
**What**: Closes #6383

### CODE REVIEW

**Review by kmcfaul — APPROVED** [review]

File: packages/react-core/src/components/Select/SelectOption.tsx [hunk 1]
```diff
@@ -228,166 +231,209 @@ export class SelectOption extends React.Component<SelectOptionProps> {
...
-                          )}
-                        </React.Fragment>
-                      )}
-                    </Component>
-                    {isFavorite !== null && id && favoriteButton(onFavorite)}
-                  </>
-                )}
-              </li>
+    const renderOption = (onSelect: any, onClose: any, variant: string, inputIdPrefix: string, onFavorite: any) => {
+      if (variant !== SelectVariant.checkbox && isLoading && isGrouped) {
```
Comment (nicolethoen) [inline packages/react-core/src/components/Select/SelectOption.tsx:235]: Could you add a one line comment to describe the situation for each of these conditionals?

File: packages/react-core/src/components/Select/SelectOption.tsx [hunk 2]
```diff
@@ -228,166 +231,209 @@ export class SelectOption extends React.Component<SelectOptionProps> {
...
+              !isLoad && !isLoading && styles.selectMenuWrapper,
+              isFavorite && styles.modifiers.favorite,
+              isFocused && styles.modifiers.focus,
+              isLoading && styles.modifiers.loading
             )}
-            {variant === SelectVariant.checkbox && isLoad && (
+            ref={this.liRef}
+          >
+            {isLoading && children}
+            {/* {isLoading && !isGrouped && children} */}
```
Comment (nicolethoen) [inline packages/react-core/src/components/Select/SelectOption.tsx:277]: This can be cleaned up

File: packages/react-core/src/components/Select/SelectOption.tsx [hunk 3]
```diff
@@ -228,166 +231,209 @@ export class SelectOption extends React.Component<SelectOptionProps> {
...
-                            </span>
-                          )}
-                        </React.Fragment>
-                      )}
-                    </Component>
-                    {isFavorite !== null && id && favoriteButton(onFavorite)}
-                  </>
-                )}
-              </li>
+    const renderOption = (onSelect: any, onClose: any, variant: string, inputIdPrefix: string, onFavorite: any) => {
```
[the remark this answers is not in the document]
Comment (tlabaj) [reply packages/react-core/src/components/Select/SelectOption.tsx:234]: Good catch I meant to update those after I verified fix.

File: packages/react-core/src/components/Select/SelectOption.tsx (hunk 1 above)
Comment (tlabaj) [reply packages/react-core/src/components/Select/SelectOption.tsx:235]: I don't know that that is necessary.  I just copied what was there before and move it into a function.  the comment would also pretty much just state what is in the if statement.

Comment (nicolethoen) [reply packages/react-core/src/components/Select/SelectOption.tsx:245]: okay - I'm just imagining future scenarios where we want to make a small change, and trying to figure out what might be the side effects or what other logic scenarios exist would be made a lot easier if they were described in plain english a bit?

File: packages/react-integration/demo-app-ts/src/components/demos/SelectDemo/SelectViewMoreGroupedDemo.tsx [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+  isOpenCheck: boolean;
+  selected: string[];
+  selectedCheck: string[];
+  numOptions: number;
+  numOptionsCeck: number;
+  isLoading: boolean;
+  isLoadingCheck: boolean;
+}
+
+export class SelectViewMoreGroupedDemo extends Component<SelectViewMoreGroupedDemoState> {
```
[the remark this answers is not in the document]
Comment (jeffpuzzo) [reply packages/react-integration/demo-app-ts/src/components/demos/SelectDemo/SelectViewMoreGroupedDemo.tsx:24]: Do we plan on migrating all components to functional ones? If so, I wonder if a functional one should be written instead. I'll leave it up to you though.

[the remark this answers is not in the document]
Comment (tlabaj) [reply packages/react-integration/demo-app-ts/src/components/demos/SelectDemo/SelectViewMoreGroupedDemo.tsx:24]: The plan is to remove the integration app one our react core demos are converted to typescript. I don't think it is worth the effort to convert the demo's.

**Review by jeffpuzzo — APPROVED** [review]

**Review by nicolethoen — APPROVED** [review]
